examples/native-external-references-mockshop/functions/mockShop.ts

```diff
@@ -0,0 +1,181 @@
+import { FunctionEventContext } from '@contentful/functions-types';
+import {
+  EventHandler,
+  MappingHandler,
+  Product,
+  ProductLookupData,
+  QueryHandler,
+  ResourcesLookupHandler,
+  ResourcesSearchHandler,
+  SearchResultData,
+} from './types';
+
+const getMockShopUrl = (context: FunctionEventContext<Record<string, any>>) => {
+  const { apiEndpoint } = context.appInstallationParameters;
+  let mockShopUrl = apiEndpoint;
+  if (!mockShopUrl) {
+    mockShopUrl = 'https://mock.shop/api';
+    console.warn(`No API url configured, falling back to '${mockShopUrl}'`);
+  }
+  return mockShopUrl;
+};
+
+function withUrn(node: Product) {
+  return {
+    ...node,
+    urn: node.id,
+  };
+}
+
+function withBadge(node: Product) {
+  return {
+    ...node,
+    badge: { variant: 'primary', label: 'it works' },
+  };
+}
+
+const resourceTypeMappingHandler: MappingHandler = (event) => {
+  const mappings = event.resourceTypes.map(({ resourceTypeId }) => ({
+    resourceTypeId,
+    graphQLOutputType: 'Product',
+    graphQLQueryField: 'product',
+    graphQLQueryArguments: { id: '/urn' },
+  }));
+
+  return {
+    resourceTypes: mappings,
+  };
+};
+
+const queryHandler: QueryHandler = async (event, context) => {
+  // Installation parameters are defined in the app definition
+  // and set per installation.
+
+  const mockShopUrl = getMockShopUrl(context);
+
+  // Make a request to the third party API.
+  // The expected return type aligns with the
+  // one outlined in the GraphQL specs:
+  // https://spec.graphql.org/October2021/#sec-Response
+  const response = await fetch(mockShopUrl, {
+    body: JSON.stringify({
+      query: event.query,
+      operationName: event.operationName,
+      variables: event.variables,
+    }),
+    method: 'POST',
+    headers: { Accept: 'application/json', 'content-type': 'application/json' },
+  });
+
+  return response.json();
+};
+
+const searchHandler: ResourcesSearchHandler = async (event, context) => {
+  const { query, resourceType } = event;
+  const mockShopUrl = getMockShopUrl(context);
+
+  if (resourceType !== 'MockShop:Product') {
```

Comment:
I would remove this from the tutorial as this should never happen so the developer does not need to defensively code around that preemptively